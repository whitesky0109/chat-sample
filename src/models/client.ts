import { UserPayload } from 'public/store/actions/userAction';
import { RoomPayload } from 'public/store/actions/roomAction';

export interface IMessage {
  roomId?: string;
  type: string;
  message: File | string;
}

export interface StoreState {
  user: UserPayload;
  room: RoomPayload;
}
