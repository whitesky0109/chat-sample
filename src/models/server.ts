export interface IMigration {
  migration_id: string;
  action: string;
  args: any[];
}

export interface IRepository {
  migrations(): IMigration[];
}

export interface IService {
  init(): Promise<any>;
}

export interface IMigrationLog {
  id: number;
  migration_id: string;
  action: string;
  success: boolean;
  error: string;
  timestamp: Date;
}

export interface IMigrationLogs {
  // tslint:disable-next-line: variable-name
  [migration_id: string]: IMigrationLog;
}

export interface IInvite {
  roomId: string;
  users: string[];
}

export interface IMessage {
  userId: string;
  type: string;
  message: ArrayBuffer | string;
  date: Date;
  isMy: boolean;
}

export interface IUser {
  [userId: string]: string | null;
}

export interface IRoomInfo {
  name: string;
  users: IUser;
  creator: string;
}

export interface IRoom {
  [roomId: string]: IRoomInfo;
}
