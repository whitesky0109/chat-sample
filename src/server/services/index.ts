import 'reflect-metadata';
// System Services
import {
    LoggerService,
    SocketManagerService,
} from './system';

// Chat Service
import {
    UserService,
    RoomService,
} from './chat';

export {
    SocketManagerService,
    LoggerService,
    UserService,
    RoomService,
};
