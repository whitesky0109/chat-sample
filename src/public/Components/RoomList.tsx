import _ from 'lodash';
import React, { Component, ReactNode } from 'react';
import { IRoom } from '../../models/server';

export interface Props {
  rooms: IRoom;
  onRoomClick: (roomId: string) => void;
}

export default class RoomList extends Component<Props> {

  onClickItem = (roomId: string) => {
    const { onRoomClick } = this.props;

    if (onRoomClick) {
      onRoomClick(roomId);
    }
  }

  renderList(): ReactNode {
    const { rooms } = this.props;
    return _.map(rooms, (room, roomId: string) => {
      return (<li className="user clearfix" key={roomId} onClick={this.onClickItem.bind(this, roomId)}>
        <div className="user-icon">
          <i className="fa fa-users" />
        </div>
        <div className="about">
          <div className="name">{room.name}</div>
          <div className="status">
            <i className="fa fa-user" />
            <span> created by {room.creator}</span>
          </div>
        </div>
      </li>);
    });
  }

  render() {
    const { rooms } = this.props;
    if (!rooms || rooms === {}) {
      return <></>;
    }

    return <div className="users">
      <div className="empty" />
      <ul className="list">
        {this.renderList()}
      </ul>
    </div>;
  }
}
