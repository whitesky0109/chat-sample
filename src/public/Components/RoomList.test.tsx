import React from 'react';
import { shallow } from 'enzyme';
import RoomList from './RoomList';
import { IRoom } from 'models/server';

describe('채팅방 목록', () => {
  let component: any = null;
  const mockOnRoomClick = jest.fn();
  const room: IRoom = {
    ['roomid1']: {
      creator: 'userid',
      name: 'roomname',
      users: {
        ['userid1']: 'sockid',
        ['userid2']: null,
      },
    },
  };

  it('renders correctly', () => {
    component = shallow(<RoomList rooms={room} onRoomClick={mockOnRoomClick} />);
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('채팅방 이름 확인', () => {
    const div = component.find('.name');
    expect(div.at(0).text()).toBe('roomname');
  });

  it('채팅방 생성자 이름 확인', () => {
    const span = component.find('.status>span');
    expect(span.at(0).text()).toBe(' created by userid');

  });

  it('입력 이벤트 함수 동작 확인', () => {
    const li = component.find('li');
    li.at(0).simulate('click');
    expect(mockOnRoomClick.mock.calls.length).toBe(1);
  });

});
