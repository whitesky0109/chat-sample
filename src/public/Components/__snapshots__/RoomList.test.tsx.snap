// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`채팅방 목록 matches snapshot 1`] = `
<div
  className="users"
>
  <div
    className="empty"
  />
  <ul
    className="list"
  >
    <li
      className="user clearfix"
      key="roomid1"
      onClick={[Function]}
    >
      <div
        className="user-icon"
      >
        <i
          className="fa fa-users"
        />
      </div>
      <div
        className="about"
      >
        <div
          className="name"
        >
          roomname
        </div>
        <div
          className="status"
        >
          <i
            className="fa fa-user"
          />
          <span>
             created by 
            userid
          </span>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
